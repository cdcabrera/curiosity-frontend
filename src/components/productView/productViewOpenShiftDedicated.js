import React from 'react';
import { ProductView } from './productView';
import { ToolbarFieldRangedMonthly } from '../toolbar/toolbarFieldRangedMonthly';
import { SelectPosition } from '../form/select';

/**
 * An OpenShift Dedicated configured view.
 *
 * @returns {Node}
 */
const ProductViewOpenShiftDedicated = () => (
  <ProductView toolbarGraph={<ToolbarFieldRangedMonthly position={SelectPosition.right} />} toolbarGraphDescription />
);

/**
 * Prop types.
 *
 * @type {{}}
 */
ProductViewOpenShiftDedicated.propTypes = {};

/**
 * Default props.
 *
 * @type {{}}
 */
ProductViewOpenShiftDedicated.defaultProps = {};

export { ProductViewOpenShiftDedicated as default, ProductViewOpenShiftDedicated };
