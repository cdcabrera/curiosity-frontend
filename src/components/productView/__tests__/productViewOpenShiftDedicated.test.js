import React from 'react';
import { shallow } from 'enzyme';
import { ProductViewOpenShiftDedicated } from '../productViewOpenShiftDedicated';
import { config } from '../../../config/product.openshiftDedicated';

describe('ProductViewOpenShiftDedicated Component', () => {
  it('should render a non-connected component', () => {
    const props = {
      routeDetail: {
        pathParameter: 'lorem ipsum',
        productConfig: [config],
        productParameter: 'dolor sit'
      }
    };

    const component = shallow(<ProductViewOpenShiftDedicated {...props} />);
    expect(component).toMatchSnapshot('non-connected');
  });
});
