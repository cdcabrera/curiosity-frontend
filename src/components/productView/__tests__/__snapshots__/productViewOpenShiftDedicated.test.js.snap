// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductViewOpenShiftDedicated Component should render a non-connected component: non-connected 1`] = `
<ProductView
  routeDetail={
    Object {
      "pathParameter": "lorem ipsum",
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "coreHours",
              "stroke": "#06c",
            },
          ],
          "initialGraphSettings": Object {
            "actionDisplay": [Function],
          },
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "coreHours",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"lastSeen_OpenShift-dedicated-metrics\\"})",
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": undefined,
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "query": Object {
            "beginning": "2019-07-01T00:00:00.000Z",
            "ending": "2019-07-31T23:59:59.999Z",
          },
        },
      ],
      "productParameter": "dolor sit",
    }
  }
  t={[Function]}
  toolbarGraph={
    <ToolbarFieldRangedMonthly
      options={
        Array [
          Object {
            "_title": "current",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"current\\"})",
            "value": Object {
              "endDate": 2019-07-31T23:59:59.999Z,
              "startDate": 2019-07-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "june",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"June\\"})",
            "value": Object {
              "endDate": 2019-06-30T23:59:59.999Z,
              "startDate": 2019-06-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "may",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"May\\"})",
            "value": Object {
              "endDate": 2019-05-31T23:59:59.999Z,
              "startDate": 2019-05-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "april",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"April\\"})",
            "value": Object {
              "endDate": 2019-04-30T23:59:59.999Z,
              "startDate": 2019-04-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "march",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"March\\"})",
            "value": Object {
              "endDate": 2019-03-31T23:59:59.999Z,
              "startDate": 2019-03-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "february",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"February\\"})",
            "value": Object {
              "endDate": 2019-02-28T23:59:59.999Z,
              "startDate": 2019-02-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "january",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"January\\"})",
            "value": Object {
              "endDate": 2019-01-31T23:59:59.999Z,
              "startDate": 2019-01-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "december",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"December 2018\\"})",
            "value": Object {
              "endDate": 2018-12-31T23:59:59.999Z,
              "startDate": 2018-12-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "november",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"November 2018\\"})",
            "value": Object {
              "endDate": 2018-11-30T23:59:59.999Z,
              "startDate": 2018-11-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "october",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"October 2018\\"})",
            "value": Object {
              "endDate": 2018-10-31T23:59:59.999Z,
              "startDate": 2018-10-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "september",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"September 2018\\"})",
            "value": Object {
              "endDate": 2018-09-30T23:59:59.999Z,
              "startDate": 2018-09-01T00:00:00.000Z,
            },
          },
          Object {
            "_title": "august",
            "selected": false,
            "title": "t(curiosity-toolbar.granularityRange, {\\"context\\":\\"August 2018\\"})",
            "value": Object {
              "endDate": 2018-08-31T23:59:59.999Z,
              "startDate": 2018-08-01T00:00:00.000Z,
            },
          },
        ]
      }
      t={[Function]}
      value="t(curiosity-toolbar.granularityRange, {\\"context\\":\\"current\\"})"
      viewId="toolbarFieldRangeGranularity"
    />
  }
  toolbarGraphDescription={true}
  toolbarProduct={false}
/>
`;
